timeout.go

```diff
@@ -61,26 +65,31 @@ func NewTimerWheel(min, max time.Duration) *TimerWheel {
 	// timeout
 	wLen := int((max / min) + 2)
 
-	tw := TimerWheel{
+	tw := TimerWheel[T]{
```

Comment:
nit: I didn't like the fact that `lastTick` isn't initialized until Advance() is called. The net effect is that delays of events placed in the timer wheel don't take effect until after the first Advance() is called. So, if I place a 10s event in the wheel, and then call Advance after 5s, the event doesn't Purge until 15s after it was initially placed in the wheel.